File: frontend/src/api/seasons.ts
import { apiRequest } from "./client";

export interface EpisodeDetail {
  id: string; title: string; index: number | null; index_end: number | null;
  image: string | null; summary: string | null; aired: string | null;
  runtime_minutes: number | null; rating: number | null;
  directors: string[]; writers: string[]; cast: string[];
}
export interface SeasonDetail {
  id: string; series_id: string; title: string; index: number | null;
  poster: string | null; background: string | null; summary: string | null;
  episodes: EpisodeDetail[];
}
export const seasonsApi = {
  get: (server: number, series: string, season: string) => apiRequest<SeasonDetail>(`/servers/${server}/shows/${encodeURIComponent(series)}/seasons/${encodeURIComponent(season)}`),
};

File: crates/posterview-contracts/src/seasons.rs
use serde::Serialize;

#[derive(Debug, Clone, Serialize)]
pub struct EpisodeDetail {
    pub id: String,
    pub title: String,
    pub index: Option<i64>,
    pub index_end: Option<i64>,
    pub image: Option<String>,
    pub summary: Option<String>,
    pub aired: Option<String>,
    pub runtime_minutes: Option<i64>,
    pub rating: Option<f64>,
    pub directors: Vec<String>,
    pub writers: Vec<String>,
    pub cast: Vec<String>,
}

#[derive(Debug, Clone, Serialize)]
pub struct SeasonDetail {
    pub id: String,
    pub series_id: String,
    pub title: String,
    pub index: Option<i64>,
    pub poster: Option<String>,
    pub background: Option<String>,
    pub summary: Option<String>,
    pub episodes: Vec<EpisodeDetail>,
}
